src/alert-fastlog.c

```diff
@@ -141,24 +141,27 @@ int AlertFastLogger(ThreadVars *tv, void *data, const Packet *p)
             action = "[wDrop] ";
         }
 
-        char proto[16] = "";
+        /* Create the alert string without locking. */
+        int size = 0;
         if (likely(decoder_event == 0)) {
+            char proto[16] = "";
             if (SCProtoNameValid(IP_GET_IPPROTO(p)) == TRUE) {
                 strlcpy(proto, known_proto[IP_GET_IPPROTO(p)], sizeof(proto));
             } else {
                 snprintf(proto, sizeof(proto), "PROTO:%03" PRIu32, IP_GET_IPPROTO(p));
             }
-        }
-
-        /* Create the alert string without locking. */
-        int size = 0;
-        if (likely(decoder_event == 0)) {
-            PrintBufferData(alert_buffer, &size, MAX_FASTLOG_ALERT_SIZE, 
+            uint16_t src_port_or_icmp = p->sp;
+            uint16_t dst_port_or_icmp = p->dp;
+            if (IP_GET_IPPROTO(p) == IPPROTO_ICMP) {
```

Comment:
Should this handle `IPPROTO_CIMPV6`?

Suggest using if/else:
```
    if _is_icmp_
        set src and dest to type and code, respectively
   else:
        set src/dest ports
```

Reply:
This is just a back port. Maybe we should first tackle main branch with this, don't you think ?

Reply:
Minimally, shouldn't both include `IPPROTO_ICMPV6`?

Reply:
> Should this handle `IPPROTO_CIMPV6`?

Yes it should. I'm fixing this up both here and for master.